modular_skyrat/modules/container_emotes/code/container_emotes.dm

```diff
@@ -60,11 +78,24 @@
 
 	container_message = ("[user.say_emphasis(container_message)]")
 
-	if (isturf(user.loc) || (!user.loc) || user.IsUnconscious()) //one last sanity check
+	var/atom/picked_loc
+	if (locs_we_can_use.len == 0)
+		return FALSE
+	if (locs_we_can_use.len == 1)
+		picked_loc = pick(locs_we_can_use)
+	else
+		picked_loc = tgui_input_list(user, "Which container would you like your emote to originate from?", "Container emote", locs_we_can_use, FALSE)
+	// Since the tgui input sleeps, we can no longer trust the status of any variable after this point
+	// Ex. we cannot assume the user exists anymore
+	if(!can_run_emote(user))
+		return FALSE
```

Comment:
```suggestion

	// Since the tgui input sleeps, we can no longer trust the status of any variable after this point
	// Ex. we cannot assume the user exists anymore
	if(!can_run_emote(user))
		return FALSE

```